### PR TITLE
Bug sending messages

### PR DESCRIPTION
**Pull request by connieyu18**
* Fixed sending messages bug by adding async await  to thunk action creator functions 
* Fixed displaying the correct order on page load 
* Fixed security bug related to sending messages that belongs to correct SenderId And RecipientId

### CODE REVIEW

File: client/src/store/utils/thunkCreators.js
Comment (Tahoum) [inline client/src/store/utils/thunkCreators.js:98]: 💯

File: server/routes/api/conversations.js
Comment (Tahoum) [inline server/routes/api/conversations.js]: One consequence of this fix is that now conversations are out of order! we want the conversations to stay ordered by the ones with the latest message on top.

File: server/routes/api/messages.js
Comment (Tahoum) [inline server/routes/api/messages.js]: Is this needed?

**Review by Tahoum — COMMENTED** [review]
Good work.
Please address the comments and once you're done let us know in the channel so we can give you a second review 🙂

File: server/routes/api/conversations.js
Comment (connieyu18) [reply server/routes/api/conversations.js]: Thanks for correcting me!

File: server/routes/api/messages.js
Comment (connieyu18) [reply server/routes/api/messages.js]: No we don't really need to return since it's at the end of the function

File: server/routes/api/conversations.js
Comment (Mayo19) [inline server/routes/api/conversations.js:25]: When running the app, were you able to verify this fixed/changed the ordering? Since changing line 23 like you did before also changes the order of the conversations so we'll have to find another way to solve it.

File: server/routes/api/messages.js
Comment (Mayo19) [inline server/routes/api/messages.js:44]: Might be better to send back a 403 instead. [Explaination](https://stackoverflow.com/questions/29912305/response-code-400-or-403-for-post-restful-apis)

**Review by Mayo19 — APPROVED** [review]
You found all the issues! I left a comment regarding if one of the changes actually fixes the messages display issue.

I've approved this, feel free to merge once you've addressed the comments.

File: server/routes/api/conversations.js
Comment (connieyu18) [reply server/routes/api/conversations.js:25]: Thanks for your correction. I updated by querying using updatedAt for conversations and ordering messages in asc order. I am still familiarizing myself with Sequelize.

**Pull request merged**
